src/main/java/org/isf/accounting/rest/BillController.java

```diff
@@ -0,0 +1,449 @@
+        datefrom.setTime(dateFrom);
+        
+        GregorianCalendar dateto = new GregorianCalendar();
+        dateto.setTime(dateTo);
+               
+        BillItems billItem = getObjectMapper().map(billItemDTO, BillItems.class);
+        	
+        logger.info("Get Bills datefrom:"  +  datefrom + " dateTo:" + dateto + " Bill ITEM ID: "+billItem.getId());
+             
+        ArrayList<Bill> bills = billManager.getBills(datefrom, dateto, billItem);
+        
+        List<BillDTO> billDTOS = bills.stream().map(bil-> getObjectMapper().map(bil, BillDTO.class)).collect(Collectors.toList());
+        
+        if(billDTOS.size() == 0){
+            return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+        }else{
+            return ResponseEntity.ok(billDTOS);
+        }
+	}
+	
+	/**
+	 * Returns all the distinct stored {@link BillItems}.
+	 * 
+	 * @return a list of  distinct {@link BillItems} or null if an error occurs.
+	 * @throws OHServiceException 
+	 */
+	@GetMapping(value = "/bills/items", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<List<BillItemsDTO>> getDistinctItems() throws OHServiceException {
+		
+		logger.info("get all the distinct stored BillItems");
+           
+	    ArrayList<BillItems> items = billManager.getDistinctItems();
+	    
+        List<BillItemsDTO> itemsDTOS = items.stream().map(it-> getObjectMapper().map(it, BillItemsDTO.class)).collect(Collectors.toList());
+        
+        if(itemsDTOS.size() == 0){
+            return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+        }else{
+            return ResponseEntity.ok(itemsDTOS);
+        }
+	}
+	
+	@DeleteMapping(value = "/bills/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity deleteBill(@PathVariable Integer id) throws OHServiceException {
+        logger.info("Delete bill id:"  +  id);
+        Bill bill = billManager.getBill(id);
+        boolean isDeleted = false;
+        if (bill != null) {
+            isDeleted = billManager.deleteBill(bill);
+        } else {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(null);
+        }
+        if (!isDeleted) {
+            throw new OHAPIException(new OHExceptionMessage(null, "Bill is not deleted!", OHSeverityLevel.ERROR));
+        }
+        return ResponseEntity.ok(isDeleted);
+    }
+	
+	/**
+	 * search all the {@link Bill}s associated to the passed {@link BillPayments}.
+	 * @param payments the {@link BillPayments} associated to the bill to retrieve.
+	 * @return a list of {@link Bill} associated to the passed {@link BillPayments} or <code>null</code> if an error occurred.
+	 * @throws OHServiceException
+	 */
+	@PostMapping(value = "/bills/search/by/payments", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<List<BillDTO>> searchBillsByPayments(@RequestBody ArrayList<BillPaymentsDTO> paymentsDTO) throws OHServiceException {
```

Comment:
Could we use List instead of ArrayList as parameter type?

Reply:
OK.